Feature/profile page

j-ferber opened the pull request:
Added the profile pages for users. These pages are only visible to users who are logged in. Also added a 404 page and adjusted the header a little as well.

Image of profile page:
![image](https://github.com/user-attachments/assets/ef5ea1a0-8be1-4a72-be4a-30656063b39d)

I also noticed that we have the tasksPosted array under the users with the postID's. However, this may be unnecessary as the post objects have posterID field, so we can search the posts collection for this instead of using the subarray of postIDs. This can be discussed later.

j-ferber committed:
Added profile pages for users

j-ferber committed:
Formatting

Victor-M-Giraldo commented:
I think this is good to merge.

Victor-M-Giraldo commented:
Actually, just realized. Shouldn't we respond with a 401 here?
```js
    if (!req.session.profile) {
      return res.redirect("/users/login");
    }
```

j-ferber committed:
small fixes

j-ferber commented:
I added the 401 status to the redirect. Also, I changed the profile routes to be under `/users/profile/{username}` instead of `/users/{username}` because if a user had the name "login" or "signup" it would run into issues with the normal routes for login and signup.
